daemons/fenced/fenced_history.c

```diff
@@ -240,6 +282,11 @@ stonith_merge_in_history_list(GHashTable *history)
             continue; /* skip over existant - state-merging migh be desirable */
         }
 
+        if (g_hash_table_size(stonith_remote_op_list) >= MAX_STONITH_HISTORY) {
+            stonith_fence_history_trim(TRUE);
```

Comment:
If all the nodes merge after a partition, and it brings them over the limit, I'm a little concerned about the redundancy of every node requesting a trim. But I'm not sure it can be avoided.